plugins/star-rating/api/api.js

```diff
@@ -10,30 +10,349 @@ var plugin = {},
+
+        common.db.collection(collectionName)
+            .findAndModify({_id:common.db.ObjectID(id)},{}, changes,
+                function(err, widget) {
+                    if (!err) {
+                        common.returnMessage(ob.params, 200, 'Widget updated');
+                        return true;
+                    } else {
+                        common.returnMessage(ob.params, 500, err.message);
+                        return false;
+                    }
+                })
+        return true;
+    }
+
+    var removeWidgetData = function(widgetId, app, callback) {
+        var collectionName = "feedback" + app;
+
+        common.db.collection(collectionName)
+            .remove({"widget_id":widgetId},
+            function(err) {
+                if (!err) callback(null);
+                else callback(err);
+            })
+    }
+
+    plugins.register("/i", function(ob){
+        var params = ob.params;
+        if (params.qstring.events && params.qstring.events.length && Array.isArray(params.qstring.events)) {
+            params.qstring.events = params.qstring.events.filter(function(currEvent){
+                if (currEvent.key == "[CLY]_star_rating"){
+                    
+                    var collectionName = 'feedback' + params.qstring.app_key;
+
+                    common.db.collection(collectionName)
+                        .insert({
+                            "email": currEvent.segmentation.email,
+                            "comment": currEvent.segmentation.comment,
+                            "ts":currEvent.timestamp,
+                            "device_id":params.qstring.device_id,
+                            "cd":new Date(),
+                            "uid":params.app_user.uid,
+                            "contact_me":currEvent.segmentation.contactMe,
+                            "rating":currEvent.segmentation.rating,
+                            "widget_id":currEvent.segmentation.widget_id
+                        }, function(err, saved) {
+                            /**
+                             *  register for process new  rating event data.
+                             *  the original event format like:
+                             *  { key: '[CLY]_star_rating', count:1, sum:1, segmentation:{ platform:"iOS", version:"3.2", rating:2}
+                             *  this function will add a field call "platform_version_rate" in segmentation.
+                             */
+                            currEvent.segmentation['platform_version_rate'] = currEvent.segmentation['platform'] + "**" + currEvent.segmentation['app_version'] + "**" + currEvent.segmentation['rating'] + "**";
+                            // remove data from core data
+                            // for GDPR compatibility
+                            delete currEvent.segmentation.email;
+                            delete currEvent.segmentation.comment;       
```

Comment:
Need to figure out a way to store these segments for drill, but exclude them from common events